tests/update_expected_runtime.py

```diff
@@ -0,0 +1,152 @@
+import sys
+import os
+import csv
+import argparse
+
+# global variables - files names
+run_log = 'tests/run_log.txt'
+expected_time_file = 'tests/tests_expected_runtime.csv'
+
+
+def _get_run_log_summary_lines():
+    # getting the runtime of test from the log's summary
+    with open(run_log, 'r') as f:
+        lines = f.readlines()
+    start_idx = 0
+    for idx, line in enumerate(lines):
+        if 'Passed Tests:' in line:
+            start_idx = idx + 1
+    return lines[start_idx:]
+
+
+def _get_test_name_from_line(line):
+    if 'cmdline_' in line:
+        return line.split('(')[0].rstrip()
+    else:
+        return line.split('/tests/')[1].split('(')[0].rstrip()
+
+
+def _get_test_run_time_from_line(line):
+    return line.split('(')[1].split(' ')[0]
+
+
+def _get_new_run_time(test_name):
+    lines = _get_run_log_summary_lines()
+    for line in lines:
+        if test_name in line:
+            return _get_test_run_time_from_line(line)
+
+    return ''  # should not get here
+
+
+def _update_run_time_in_row(row, new_run_time):
+    row_list = row.split(',')
+    row_list[1] = new_run_time + '\n'
+    return ','.join(row_list)
+
+
+def _update_or_delete_file_rows(lines_to_update, lines_to_delete):
+    print('Updating the runtime of modified tests and deleting the rows of deleted tests (if any)')
+    lines_to_delete.sort(reverse=True)
+    # get the lines of the tests_expected_runtime.csv file as a list
+    with open(expected_time_file, 'r') as f:
+        lines = f.readlines()
+    # update the relevant rows in the lines list
+    for line_num, test_name in lines_to_update:
+        lines[line_num-1] = _update_run_time_in_row(lines[line_num-1], _get_new_run_time(test_name))
+    # delete the rows to delete in the lines list - in descending order
+    for line_num in lines_to_delete:
+        del lines[line_num - 1]
+    # write the new lines list into the file again
+    with open(expected_time_file, 'w', newline='') as f:
+        for line in lines:
+            f.write(line)
+
+
+def _add_test_to_expected_runtime_file(test_name, run_time):
+    with open(expected_time_file, 'a', newline='') as csv_file:
+        csv_writer = csv.writer(csv_file)
+        csv_writer.writerow([test_name, run_time])
+    csv_file.close()
+
+
+def _add_cmdline_tests_to_file():
+    print('Updating cmdline tests with their new runtime')
+    lines = _get_run_log_summary_lines()
+    for line in lines:
+        if 'cmdline_' in line:
+            _add_test_to_expected_runtime_file(_get_test_name_from_line(line), _get_test_run_time_from_line())
+
+
+def _update_tests_runtime(modified_tests_list):
```

Comment:
add some documentation about the current limitation of update when a test is renamed, due to output from 'git diff' in the workflow run